### PR TITLE
Upgrades React 

### PR DESCRIPTION
**Pull request by orenf**
Upgrades React to version 18

Adds @testing-library/react

Updates tests accordingly

### CODE REVIEW

File: packages/core/test/Form.test.jsx [hunk 1]
```diff
@@ -2,8 +2,9 @@ import * as React from 'react';
 import { expect } from 'chai';
 import sinon from 'sinon';
 import { createRef } from 'react';
-import { renderIntoDocument, act, Simulate } from 'react-dom/test-utils';
-import { render, findDOMNode } from 'react-dom';
+import { fireEvent, act, render } from '@testing-library/react';
+import { Simulate } from 'react-dom/test-utils';
```
Comment (heath-freenome) [inline packages/core/test/Form.test.jsx:6]: Simulate is still being used?

File: packages/core/test/ArrayField.test.jsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { expect } from 'chai';
-import { Simulate, act } from 'react-dom/test-utils';
+import { Simulate } from 'react-dom/test-utils';
```
Comment (heath-freenome) [inline packages/core/test/ArrayField.test.jsx:2]: Where is `Simulate` still being used?

File: packages/core/test/ArrayField.test.jsx [hunk 3]
```diff
@@ -1294,14 +1337,16 @@ describe('ArrayField', () => {
...
-        Simulate.change(node.querySelector('.field select'), {
-          target: {
-            options: [
-              { selected: true, value: 0 }, // use index
-              { selected: true, value: 1 }, // use index
-              { selected: false, value: 2 }, // use index
-            ],
-          },
+        act(() => {
+          Simulate.change(node.querySelector('.field select'), {
```
Comment (heath-freenome) [inline packages/core/test/ArrayField.test.jsx:1341]: Can you make this use `fireEvent` instead?

Comment (heath-freenome) [inline packages/core/test/ArrayField.test.jsx:1341]: Can you make this use `fireEvent` instead?
```suggestion
          fireEvent.change(node.querySelector('.field select'), {
```

File: packages/core/test/Form.test.jsx (hunk 1 above)
Comment (orenf) [reply packages/core/test/Form.test.jsx:6]: Thanks for taking the time to review @heath-freenome.  There are two cases where _fireEvent_ did not behave as I had expected and that's why I continued to use Simulate.  

- In the case of a **select** component with the option to select multiple values. I simply couldn't figure out how to set the  **selected** attribute on more than one option element.  

- In the case of an input field of type number. fireEvent will not set a value to a non numeric.  

I'm happy to take any suggestions.

**Comment by heath-freenome** [comment]
@orenf your tests are failing and this needs a rebase. Are you able to fix these?

**Comment by orenf** [comment]
@heath-freenome , you might be missing the second commit that I added. The second commit will fix the failing test that I believe you're referencing. I just confirmed that all the tests are passing.

File: packages/core/test/Form.test.jsx (hunk 1 above)
Comment (orenf) [reply packages/core/test/Form.test.jsx:6]: Just wanted to follow up on the React upgrade. I've been developing quite a bit on top of the React 18 library and haven't had any major issues. There are peer dependency errors that I've come across in libraries that are fixed to React 17.   You'll also see the following warning:
```Warning: ReactDOM.render is no longer supported in React 18.```

**Comment by heath-freenome** [comment]
> @heath-freenome , you might be missing the second commit that I added. The second commit will fix the failing test that I believe you're referencing. I just confirmed that all the tests are passing.

@orenf  The github pipelines are failing still. Can you please fix

**Comment by orenf** [comment]
Thanks @heath-freenome , I looked the error raised by Node 18.x build. I don't see this error on Node version 20 which builds fine, so I'll need a second to dig in.

**Comment by heath-freenome** [comment]
> Thanks @heath-freenome , I looked the error raised by Node 18.x build. I've not been able to reproduce the error. I'll need some time to figure out the reason for the discrepancies.

@orenf Have you figured out how to fix the build for your PR yet?

**Comment by orenf** [comment]
> > Thanks @heath-freenome , I looked the error raised by Node 18.x build. I've not been able to reproduce the error. I'll need some time to figure out the reason for the discrepancies.
> 
> @orenf Have you figured out how to fix the build for your PR yet?

I haven't had a chance to review this issue and I likely won't have the opportunity to do so this week either. I do, however, keep rebasing against the *main* branch.

**Comment by orenf** [comment]
@heath-freenome , the build appears to be passing now.

**Review by heath-freenome — APPROVED** [review]

**Pull request merged**